src/vscode-dts/vscode.proposed.terminalExitReason.d.ts

```diff
@@ -0,0 +1,35 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	// https://github.com/microsoft/vscode/issues/130231
+
+	/**
+	 * Terminal exit reason kind.
+	 */
+	export enum TerminalExitReason {
+		/**
+		 * Unknow reason, possible reasons could be:
+		 * - Triggered by user action
+		 * - Triggered by running process
+		 * - Triggered by an extension
+		 */
+		Unknown = 0,
```

Comment:
I would have expected a little more granularity in the enum values instead of lumping everything into unknown if we had an API like this.

Reply:
Added more values `Process`, `User` , `Extension`